src/PointwiseFunctions/GeneralRelativity/SpacetimeDerivativeOfGothG.cpp

```diff
@@ -0,0 +1,114 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/TimeDerivOfSpatialMetric.hpp"
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Expressions/TensorExpression.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace gr {
+template <typename DataType, size_t SpatialDim, typename Frame>
+void spacetime_deriv_of_goth_g(
+    gsl::not_null<tnsr::aBB<DataType, SpatialDim, Frame>*> da_goth_g,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::abb<DataType, SpatialDim, Frame>& da_spacetime_metric,
+    const Scalar<DataType>& lapse,
+    const Scalar<DataType>& dt_lapse,
+    const tnsr::i<DataType, SpatialDim, Frame>& deriv_lapse,
+    const Scalar<DataType>& sqrt_det_spatial_metric,
+    const tnsr::a<DataType, SpatialDim, Frame>& da_det_spatial_metric) {
+  if (UNLIKELY(get_size(get<0, 0, 0>(*da_goth_g)) !=
+               get_size(get<0, 0, 0>(da_spacetime_metric)))) {
+    set_number_of_grid_points(da_goth_g, da_spacetime_metric);
+  }
+
+  const tnsr::a<DataType, SpatialDim, Frame> da_lapse{};
```

Comment:
Leave this non-const instead of using `const_cast` below.  Or you can use the lambda initialization trick if you really want a const variable:
```c++
const tnsr::<...> da_lapse = [&]() {
  if constexpr (...) {
    // ...construct and initialize result here...
    return ...;
  } else ...
}();
```

Reply:
`make_const_view(make_not_null(&std::as_const(da_lapse.get(0))), ...`